x-pack/plugins/task_manager/task_manager.ts

```diff
@@ -84,24 +84,19 @@ export class TaskManager {
     this.store = store;
     this.poller = poller;
 
-    kbnServer.afterPluginsInit(() => {
+    kbnServer.afterPluginsInit(async () => {
+      this.isInitialized = true;
       store.addSupportedTypes(Object.keys(this.definitions));
-
-      const startStoreAndPoller = async () => {
-        try {
-          if (!this.isInitialized) {
-            await store.init();
-            await poller.start();
-            this.isInitialized = true;
-          }
-        } catch (err) {
-          this.isInitialized = false;
+      const startStoreAndPoller = () => {
+        Promise.all([store.init(), poller.start()]).catch((err: Error) => {
```

Comment:
`store.init()` will throw an exception if it has already been initialized, which means that if it succeeds, but `poller` fails (unlikely looking at its code), then this can never succeed.

Ironically I had you add the exception for double-initializing it in the larger PR when this code wasn't setup this way, so perhaps we should revisit that logic? However, it seems like the `poller` shouldn't be started until the `store` has been started (given `poller`'s use of `store.fetchAvailableTasks`), therefore this should probably be more complex to ensure the first starts before the second.